cfgov/unprocessed/js/molecules/Notification.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+// Required polyfills for <IE9.
+require( '../modules/polyfill/query-selector' );
+require( '../modules/polyfill/class-list' );
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+
+/**
+ * Notification
+ * @class
+ *
+ * @classdesc Initializes a new Notification molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} An Notification instance.
+ */
+function Notification( element ) { // eslint-disable-line max-statements, inline-comments, max-len
+
+  var BASE_CLASS = 'm-notification';
+
+  // Bitwise flags for the state of this Expandable.
+  var SUCCESS = 'success';
+  var WARNING = 'warning';
+  var ERROR = 'error';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'Notification' );
+  var _icon = _dom.querySelector( '.' + BASE_CLASS + '_icon' );
+  var _content = _dom.querySelector( '.' + BASE_CLASS + '_content' );
+
+  var _currentType;
+
+  /**
+   * @returns {Object} The Notification instance.
+   */
+  function init() {
+    // Check and set default type of notification.
+    var classList = _dom.classList;
+    if ( classList.contains( BASE_CLASS + '__' + SUCCESS ) ) {
+      _currentType = SUCCESS;
+    } else if ( classList.contains( BASE_CLASS + '__' + WARNING ) ) {
+      _currentType = WARNING;
+    } else if ( classList.contains( BASE_CLASS + '__' + ERROR ) ) {
+      _currentType = ERROR;
+    }
+
+    return this;
+  }
+
+  /**
+   * @param {number} type The notifiation type.
+   * @param {string} messageText The content of the notifiation message.
+   * @param {string|HTMLNode} explanationText
+   *   The content of the notifiation explanation.
+   * @returns {Object} The Notification instance.
+   */
+  function setContent( type, messageText, explanationText ) {
+    _setType( type );
+    var content = '<p class="h4">' +
+                  messageText +
+                  '</p>';
+    if ( typeof explanationText !== 'undefined' ) {
+      content += '<p class="h4 m-notification_explanation">' +
+                 explanationText +
+                 '</p>';
+    }
+    _content.innerHTML = content;
+
+    return this;
+  }
+
+  /**
+   * @param {number} type The notifiation type.
+   * @returns {Object} The Notification instance.
+   */
+  function _setType( type ) {
+    // If type hasn't changed, return.
+    if ( _currentType === type ) {
+      return this;
+    }
+
+    var classList = _dom.classList;
+    classList.remove( BASE_CLASS + '__' + SUCCESS );
```

Comment:
The rationale I used for the constants here are:
`BASE_CLASS` = class used in the DOM, which is named via a project-level convention that very likely may be updated.
`SUCCESS` = parameter passed into the public methods, so `_notification.setTypeAndContent( _notification.SUCCESS, 'msg' );` can be used and there's no need to lookup/accidentally misspell the flag passed to the method.

The `__` however is part of BEM. Since that is in wider use than a project-level convention it's less likely to change. If it does there would be wider implications for the codebase. So I'd say in that case having it in a constant seems like overkill (unless it's for clarity—like your `VISIBLE_CLASS` comment).